### PR TITLE
Bulk controller

### PR DESCRIPTION
**Pull request by berthieresteban**
add the bulk controller

https://github.com/kuzzleio/sdk-jvm/issues/14

### CODE REVIEW

File: doc/1/controllers/bulk/write/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+code: true
+type: page
+title: Write
```
Comment (Aschen) [inline doc/1/controllers/bulk/write/index.md:4]:
```suggestion
title: write
```

File: doc/1/controllers/bulk/write/index.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+---
+
+# Write
+
+Create or replace a document directly into the storage engine.
+
+:::: tabs
+::: tab Java
+
+## Arguments
```
Comment (Aschen) [inline doc/1/controllers/bulk/write/index.md:15]: Duplicated section title

File: doc/1/controllers/bulk/import-data/index.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
+---
+code: true
+type: page
+title: ImportData
```
[the remark this answers is not in the document]
Comment (berthieresteban) [reply doc/1/controllers/bulk/import-data/index.md:4]: In JAVA we can't call a method import so we decided to call it importData

File: doc/1/controllers/bulk/m-write/index.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+<<< ./snippets/mwrite-java.java
+
+:::
+::: tab Kotlin
+
+## Arguments
+
+```kotlin
+fun importData(
```
[the remark this answers is not in the document]
Comment (berthieresteban) [reply doc/1/controllers/bulk/m-write/index.md:78]: In JAVA we can't call a method import so we decided to call it importData

File: doc/1/controllers/bulk/m-write/index.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  - add [kuzzle metadata](/core/2/guides/essentials/document-metadata),
+  - trigger [realtime notifications](/core/2/guides/essentials/real-time) (unless asked otherwise)
+
+:::: tabs
+::: tab Java
+
+## Arguments
+
+```java
+public CompletableFuture<ConcurrentHashMap<String, Any?>> importData (
```
[the remark this answers is not in the document]
Comment (berthieresteban) [reply doc/1/controllers/bulk/m-write/index.md]: same

File: doc/1/controllers/bulk/import-data/index.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+description: Creates, updates or deletes large amounts of documents as fast as possible.
+---
+
+# ImportData
```
[the remark this answers is not in the document]
Comment (berthieresteban) [reply doc/1/controllers/bulk/import-data/index.md:8]: same

File: doc/1/controllers/bulk/delete-by-query/index.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+
+## Return
+
+An Int containing the number of deleted documents.
+
+## Exceptions
+
+Throws a `KuzzleException` if there is an error. See how to [handle errors](/sdk/csharp/2/essentials/error-handling).
+
```
Comment (jenow) [inline doc/1/controllers/bulk/delete-by-query/index.md:51]:
```suggestion
```

File: doc/1/controllers/bulk/delete-by-query/index.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
```
Comment (jenow) [inline doc/1/controllers/bulk/delete-by-query/index.md:87]:
```suggestion
```

File: doc/1/controllers/bulk/import-data/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Aschen) [reply doc/1/controllers/bulk/import-data/index.md:8]: The first letter shouldn't be uppercase so

File: doc/1/controllers/bulk/import-data/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Aschen) [reply doc/1/controllers/bulk/import-data/index.md:4]: The first letter shouldn't be uppercase so

**Review by jenow — APPROVED** [review]

**Review by Yoann-Abbes — APPROVED** [review]

**Review by Aschen — APPROVED** [review]

**Pull request merged**